docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Clustering
+weight: 500
+labels:
+  stage: beta
+---
+
+# Clustering (beta)
+
+Clustering enables a fleet of agents to work together for workload distribution
+and high availability. It helps create horizontally scalable deployments with
+minimal resource and operational overhead.
+
+To achieve this, Grafana Agent makes use of an eventually consistent model that
+assumes all participating Agents are interchangeable and converge on using the
+same configuration file.
+
+The behavior of a standalone, non-clustered agent is the same as if it was a
+single-node cluster.
+
+Clustering is configured by passing `cluster` command-line flags to the [run][]
+command.
+
+[run]: {{< relref "../reference/cli/run.md#clustering-beta" >}}
+
+## Use cases
+
+### Target auto-distribution
+
+Target auto-distribution is the most basic use case of clustering; it allows
+scraping components running on all peers to distribute scrape load between
+themselves. All nodes must have access to the same service discovery APIs, and
+the set of targets should converge on a timeline comparable to the scrape
+interval.
```

Comment:
> the set of targets should converge on a timeline comparable to the scrape interval

- This makes is sound that the time to converge goes up and down depending on the length of the scrape interval that is configured. 
- Also, it's not clear whose responsibility it is to make sure this is the case - does the Agent do it automatically, or is there a config option to make sure the converge interval is similar?

Reply:
i've reworked this sentence a bit and dropped the line about converging the targets. I have a rough idea of what Paschalis is trying to mention here, but I think convergence usually happens fast enough, and scrape intervals are usually high enough, that we don't need to document anything just yet.